File: Corrections/php/Neufplate/src/Event/EmailNotificationListener.php
<?php

namespace App\Event;

use App\Nft;
use App\User\User;
use SplObserver;
use SplSubject;

class EmailNotificationListener implements SplObserver
{

    private User $user;
    private Nft $nft;

    /**
     * @param User $user
     * @param Nft $nft
     */
    public function __construct(User $user, Nft $nft)
    {
        $this->user = $user;
        $this->nft = $nft;
    }


    public function update(SplSubject $subject): void
    {
        echo "Mail à " . $this->user->email . ": Voici votre certificat de propriété : " . $this->nft->nonce . PHP_EOL;

    }
}

File: Corrections/java/Neufplate/src/main/java/fr/neufplate/Event/EmailNotificationListener.java
package fr.neufplate.Event;

import fr.neufplate.Nft;
import fr.neufplate.User.User;

public class EmailNotificationListener implements EventListener {

    @Override
    public void update(EventType eventType, User user, String message) {
        System.out.println("\nEmail to " + user.email + " : \n" + message);
    }
}
